lib/std/fmt.zig

```diff
@@ -845,19 +845,25 @@ fn formatSizeImpl(comptime radix: comptime_int) type {
                 1024 => mags_iec[magnitude],
                 else => unreachable,
             };
-
-            try formatFloatDecimal(new_value, options, writer);
-
+            var fmt_buffer: [64]u8 = undefined; // 64bit int can't be formatted greater than 64 chars
```

Comment:
This isn't correct and can lead to a crash on input like `{:.65}`, is there a reason you can't directly use the provided `writer`?

Reply:
You need to know the size of the formatted decimal before you can work out the padding for the centre and right padding.
Otherwise you'd need to run the formatter twice, once in a dry run and a second time to actually format it. And that looked like a much larger scope than I think intended.